internal/provider/kubernetes/gateway.go

```diff
@@ -65,6 +67,11 @@ func newGatewayController(mgr manager.Manager, cfg *config.Server, su status.Upd
 	}
 	r.log.Info("watching gateway objects")
 
+	// Watch Services owned by managed Gateways.
```

Comment:
We'll also need some code that takes the Service's LB IP (once present) and populates each Gateway's `status.addresses` field with it, I'm not seeing that in this iteration.

Reply:
@skriss understood. f8fd0fa adds the logic for populating `status.addresses` with ingress ip's/hostnames from the gateway's associated service.